rom/dev/src/flow_val_rom.rs

```diff
@@ -0,0 +1,63 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    mod.rs
+
+Abstract:
+
+    File contains the implementation of the val ROM reset flows
+
+--*/
+
+#[allow(dead_code)]
+#[path = "flow/cold_reset/fw_processor.rs"]
+mod fw_processor;
+
+use crate::fht;
+use crate::{cprintln, rom_env::RomEnv};
+use caliptra_common::FirmwareHandoffTable;
+use caliptra_common::RomBootStatus::*;
+use caliptra_drivers::{report_boot_status, CaliptraResult, ResetReason};
+use caliptra_error::CaliptraError;
+use fw_processor::FirmwareProcessor;
+
+/// Execute ROM Flows based on reset reason
+///
+/// # Arguments
+///
+/// * `env` - ROM Environment
+pub fn run(env: &mut RomEnv) -> CaliptraResult<Option<FirmwareHandoffTable>> {
```

Comment:
We had discussed having a different reason other than Cold/Update/Warm Reset to trigger this validation flow. Is that not in plan?

Reply:
You are right, that was the initial plan, and I was personally the one to propose it. However while implementing Nick realized that it would require either a much more "invasive" change (or a lot of duplication) . That`s because the Mailbox APIs are private to the coldreset flow.
Do you see any issues with this "new" approach ?

Reply:
The issue of the lower-level code being private is largely resolved by the introduction of the val-specific verifier environment (we no longer had to skip the verify call as we originally thought, it's bypassed in a different way).

So an approach that adds a new flow is technically possible. But that would require touching and adding this feature switch in more of the existing ROM code files. This current approach let us eliminate changes to any rom code outside of main.rs as well as limit what needed to be touched in that file.